src/test/selftest-parallel

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+if test "$NUM_PARTITIONS" = ''; then
+    NUM_PARTITIONS=1
+fi
+if test "$RUN_PARTITIONS" = ''; then
+    RUN_PARTITIONS=$(seq 0 $((NUM_PARTITIONS-1)));
+fi
+if test "$TEST_SPEC" = ''; then
+    TEST_SPEC="~[.]" # All non-hidden tests by default
+fi
+
+cleanup() {
+    rm -rf $TEST_PARTITIONS_DIR
+}
+TEST_PARTITIONS_DIR=$(mktemp -d ./test-partitions.XXXXXXXX)
+trap cleanup 0 2 15
+
+cd $TEST_PARTITIONS_DIR
+perl -e 'my @all_tests = `../stellar-core --ll FATAL --test --list-test-names-only '"$TEST_SPEC"'`;
```

Comment:
I think it would be cleaner here to
* not `cd` into `TEST_PARTITION_DIR` and invoke stellar core with `./stellar-core` (not assuming that stellar-core can be run from a different folder - as it may load dependencies)
* pass in arguments to the perl script (such as the value of `TEST_PARTITIONS_DIR`, `NUM_PARTITIONS`)
   * this is done like this:
```sh
$ perl -e '$a=$ARGV[0]; $b=$ARGV[1]; print "a is $a; b is $b\n"' -- "A" "B's value"
a is A; b is B's value
```